### PR TITLE
Continue porting period_helper to cython

### PR DESCRIPTION
**Pull request by jbrockmendel**
This moves functions to cython but retains their C names to make comparison easier.  The next pass will get rid of the camelCase.  A couple of redundant functions are left behind in period_helper, will be cleaned up in a follow-up.

The next hurdle is to get rid of all the adding and subtracting of ORD_OFFSET, WEEK_OFFSET, BDAY_OFFSET.  (The period_helper code puts 0 at Jan 1 0001 instead of Jan 1 1970).  This would be straightforward but I want to be careful about C division/mod conventions.

### CODE REVIEW

File: pandas/_libs/tslibs/period.pyx [hunk 1]
```diff
@@ -129,6 +146,285 @@ cdef char* c_strftime(date_info *dinfo, char *fmt):
...
+    toDaily = get_asfreq_func(freq, FR_DAY)
+    get_asfreq_info(freq, FR_DAY, 'E', &af_info)
+    return toDaily(period_ordinal, &af_info) + ORD_OFFSET
+
+
+cdef int dInfoCalc_SetFromAbsDateTime(date_info *dinfo,
+                                      int64_t absdate, double abstime) nogil:
+    """
+    Set the instance's value using the given date and time.
+    Assumes GREGORIAN_CALENDAR.
```
Comment (jreback) [inline pandas/_libs/tslibs/period.pyx]: ideally add some full doc-strings on these which list the args and types

**Review by jreback — CHANGES_REQUESTED** [review]
can you run perf on periods

File: pandas/_libs/tslibs/period.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/period.pyx]: Will do.  Mind doing those in the next round to avoid clogging the CI?

**Comment by jbrockmendel** [comment]
```
taskset 5 asv continuous -f 1.1 -E virtualenv master HEAD -b period -b period -b timeseries
[...]
   before     after       ratio
  [b8351277] [769feee1]
+   19.37μs    23.65μs      1.22  period.PeriodUnaryMethods.time_now('M')
+    2.01ms     2.40ms      1.19  timeseries.ResampleSeries.time_resample('period', '5min', 'mean')
+  133.42ms   154.58ms      1.16  timeseries.DatetimeIndex.time_to_pydatetime('tz_aware')
+    7.40μs     8.44μs      1.14  timeseries.AsOf.time_asof_single_early('Series')
+    2.38ms     2.71ms      1.14  timeseries.ResampleSeries.time_resample('period', '5min', 'ohlc')
+    4.27μs     4.79μs      1.12  timeseries.DatetimeIndex.time_get('dst')
+    2.46ms     2.76ms      1.12  timeseries.ToDatetimeCache.time_dup_seconds_and_unit(False)
-    4.53μs     4.04μs      0.89  timeseries.DatetimeIndex.time_get('tz_naive')
-   71.76μs    63.44μs      0.88  period.PeriodProperties.time_property('M', 'start_time')
-    3.16μs     2.72μs      0.86  timeseries.DatetimeIndex.time_get('repeated')

    before     after       ratio
  [b8351277] [769feee1]
+   20.97ms    25.91ms      1.24  timeseries.IrregularOps.time_add
+  886.50ns     1.01μs      1.14  period.PeriodProperties.time_property('min', 'dayofweek')
+   18.87μs    20.96μs      1.11  period.PeriodUnaryMethods.time_asfreq('M')
-   19.24μs    17.26μs      0.90  timeseries.AsOf.time_asof_single('Series')
-    4.06ms     3.64ms      0.90  timeseries.ToDatetimeISO8601.time_iso8601_nosep
-  962.36ns   852.72ns      0.89  period.PeriodProperties.time_property('min', 'dayofyear')
-  153.68ms   128.59ms      0.84  timeseries.DatetimeIndex.time_to_pydatetime('tz_aware')
-   24.75μs    20.10μs      0.81  period.PeriodUnaryMethods.time_now('M')
```

**Review by jreback — APPROVED** [review]

File: pandas/_libs/tslibs/period.pyx (hunk 1 above)
Comment (jreback) [reply pandas/_libs/tslibs/period.pyx]: that's fine

**Pull request merged**